### PR TITLE
fetch admin cluster server address from configmap verrazzano-admin-cluster

### PR DESCRIPTION
**Pull request by guoyongzhang**
# Description

fetch admin cluster server address from configmap verrazzano-admin-cluster instead of kubeadm-config

Fixes VZ-2355

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/internal/k8s/adminclusterconfig.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	APIEndpoints map[string]APIEndpoint `json:"apiEndpoints"`
+}
+
+// APIEndpoint contains the kubeadmin config information needed to access the API server
+type APIEndpoint struct {
+	AdvertiseAddress string `json:"advertiseAddress"`
+	BindPort         string `json:"bindPort"`
+}
+
+// GetAPIServerURL gets the external hURL of the API server
```
Comment (pfmackin) [inline platform-operator/internal/k8s/adminclusterconfig.go:27]: spelling hURL

File: platform-operator/internal/k8s/adminclusterconfig.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	clipkg "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterStatus contains APIEndpoint map stored in the kubeadmin config map
+type ClusterStatus struct {
+	APIEndpoints map[string]APIEndpoint `json:"apiEndpoints"`
+}
+
+// APIEndpoint contains the kubeadmin config information needed to access the API server
+type APIEndpoint struct {
```
Comment (pfmackin) [inline platform-operator/internal/k8s/adminclusterconfig.go:22]: Are we going to ask the user for a full URL with port or ask for 2 params?  Can we call this json field k8sApiServerURL or something that is clearer.  The advertiseAddress is a relic of kubeadmin which is going away and it is not very clear.

File: platform-operator/internal/k8s/adminclusterconfig.go (hunk 1 above)
Comment (guoyongzhang) [reply platform-operator/internal/k8s/adminclusterconfig.go]: changed.

File: platform-operator/internal/k8s/adminclusterconfig.go (hunk 2 above)
Comment (guoyongzhang) [reply platform-operator/internal/k8s/adminclusterconfig.go]: Something simple in the configmap like `data:
  server: httsp://someurl
`

File: platform-operator/test/integ/integ_test.go [hunk 3]
```diff
@@ -25,6 +26,7 @@ const vzMcNamespace = "verrazzano-mc"
 const prometheusSecret = "prometheus-cluster1"
 const vmiESIngest = "vmi-system-es-ingest"
 const hostdata = "testhost"
+const adminClusterConfigMap = "verrazzano-admin-cluster"
```
Comment (mgianatagh) [inline platform-operator/test/integ/integ_test.go:29]: Can the test use the constant you defined for this value?

**Review by mgianatagh — APPROVED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**